File: ch1/js/j-methods.js
import { ChatOpenAI } from '@langchain/openai';

const model = new ChatOpenAI();

const response = await model.invoke('Hi there!');
console.log(response);
// Hi!

const completions = await model.batch(['Hi there!', 'Bye!']);
// ['Hi!', 'See you!']

for await (const token of await model.stream('Bye!')) {
  console.log(token);
  // Good
  // bye
  // !
}

File: ch1/py/j-methods.py
from langchain_openai.chat_models import ChatOpenAI

model = ChatOpenAI(model="gpt-3.5-turbo")

completion = model.invoke("Hi there!")
# Hi!

completions = model.batch(["Hi there!", "Bye!"])
# ['Hi!', 'See you!']

for token in model.stream("Bye!"):
    print(token)
    # Good
    # bye
    # !
